File: src/pattern/composite/java/File.java
package pattern.composite.java;

public class File extends Entry {
    private String name;
    private int size;

    public File(String name, int size){
        this.name = name;
        this.size = size;
    }

    @Override
    public String getName() {
        return this.name;
    }

    @Override
    public int getSize() {
        return this.size;
    }

    @Override
    public void printList(String prefix) {
        System.out.println(prefix+"/"+this.getName() );
    }
}

File: src/pattern/composite/kotlin/File.kt
package pattern.composite.kotlin

class File(val _name: String, val _size: Int): Entry(){
    override fun getName(): String {
        return this._name
    }

    override fun getSize(): Int {
        return this._size
    }

    override fun printList(prefix: String) {
        println("${prefix}/${this}")
    }

}